packages/plugin-node/src/index.ts

```diff
@@ -482,27 +483,13 @@ export async function buildEntrypoint({
     await finishPromise;
   }
 
-  // Update the `functions-mainifest.json` file with this entrypoint
-  // TODO NATE: turn this into a `@vercel/plugin-utils` helper function?
-  const functionsManifestPath = join(outputDirPath, 'functions-manifest.json');
-  let functionsManifest: any = {};
-  try {
-    functionsManifest = JSON.parse(
-      await fsp.readFile(functionsManifestPath, 'utf8')
-    );
-  } catch (_err) {
-    // ignore...
-  }
-  if (!functionsManifest.version) functionsManifest.version = 1;
-  if (!functionsManifest.pages) functionsManifest.pages = {};
-  functionsManifest.pages[entrypointWithoutExtIndex] = {
-    handler: `${getFileName(LAUNCHER_FILENAME).slice(0, -3)}.launcher`,
-    runtime: nodeVersion.runtime,
+  const pages = {
+    [entrypointWithoutExtIndex]: {
```

Comment:
@AndyBitz @TooTallNate I kept this as without extension but I'm wondering if it should keep the extension. Thats how we document the `functions` key (it should match source files)

Reply:
Great question! I think this also depends on the version that is returned from the Runtime, as the Build Step behaves differently for Runtimes than it does for Builders.

Can you confirm that the right thing to do is here, @AndyBitz?

Reply:
Files that go in the `pages` directory keep their extension.

Inside of the `functions-manifest.json` file, they are then referenced by their filename, **with the extension**.

Example of the `functions-manifest.json`:

```json
{
  "version": 1,
  "pages": {
    "api/date.js": {
      "memory": 128,
      "regions": ["gru1"]
    },
    "my-api.js": {
      "memory": 128,
      "maxDuration": 3,
      "regions": ["hkg1"]
    },
    "my-go-api.bin": {
      "memory": 1024,
      "runtime": "go1.x",
      "handler": "my-go-api.bin"
    },
    "my-isr-page.js": {
      "memory": 128
    },
    "my-isr-image.svg.js": {
      "memory": 3008
    }
  }
}
```

Reply:
Thanks! Looks like this was fixed in #6988